src/commands/lambda/flare.ts

```diff
@@ -40,105 +49,170 @@ export class LambdaFlareCommand extends Command {
+      const logGroupName = `/aws/lambda/${functionName}`
+      let logStreamNames
+      try {
+        logStreamNames = await getLogStreamNames(cwlClient, logGroupName)
+      } catch (err) {
+        this.context.stderr.write(commonRenderer.renderError(`Unable to get log streams: ${err.message}`))
+
+        return 1
+      }
+      if (logStreamNames === undefined) {
+        this.context.stdout.write(
+          commonRenderer.renderSoftWarning('No CloudWatch streams were found. Logs will not be retrieved or sent.\n')
+        )
+      } else {
+        this.context.stdout.write(`\n✅ Found log streams:\n• ${logStreamNames.join('\n• ')}\n\n`)
+        for (const logStreamName of logStreamNames) {
+          let logEvents
+          try {
+            logEvents = await getLogEvents(cwlClient, logGroupName, logStreamName)
+          } catch (err) {
+            this.context.stderr.write(
+              commonRenderer.renderError(`Unable to get log events for stream ${logStreamName}: ${err.message}`)
+            )
 
-      return 0
+            return 1
+          }
+          if (logEvents === undefined) {
+            this.context.stdout.write(
+              commonRenderer.renderSoftWarning(`No CloudWatch logs found in stream ${logStreamName}. Skipping...`)
+            )
+            continue
+          }
+          logs.push([logStreamName, logEvents])
+        }
+      }
     }
 
-    // Send data to Datadog
-    const folderPath = path.join(process.cwd(), FLARE_OUTPUT_DIRECTORY)
-    const filePath = path.join(folderPath, FUNCTION_CONFIG_FILE_NAME)
-    const zipPath = path.join(folderPath, ZIP_FILE_NAME)
-    const configStrUncolored = JSON.stringify(config, undefined, 2)
+    this.context.stdout.write('\n💾 Saving files...\n')
     try {
-      await writeFile(folderPath, filePath, configStrUncolored, this.context)
-      await zipContents(filePath, zipPath)
+      // Create folders
+      const rootFolderPath = path.join(process.cwd(), FLARE_OUTPUT_DIRECTORY)
+      const logsFolderPath = path.join(rootFolderPath, LOGS_DIRECTORY)
+      if (fs.existsSync(rootFolderPath)) {
+        deleteFolder(rootFolderPath)
+      }
+      createDirectories(rootFolderPath, logsFolderPath, logs.length > 0)
+
+      // Write files
+      const configFilePath = path.join(rootFolderPath, FUNCTION_CONFIG_FILE_NAME)
+      writeFile(configFilePath, JSON.stringify(config, undefined, 2))
+      this.context.stdout.write(`${logs.length > 0 ? '• ' : ''}Saved function config to ${configFilePath}\n`)
+      for (const [logStreamName, logEvents] of logs) {
+        const logFilePath = path.join(logsFolderPath, `${logStreamName.split('/').join('-')}.csv`)
+        const data = convertToCSV(logEvents)
+        writeFile(logFilePath, data)
+        this.context.stdout.write(`• Saved logs to ${logFilePath}\n`)
+        // Sleep for 1 millisecond so OS can sort files by creation time
+        await new Promise((resolve) => setTimeout(resolve, 1))
```

Comment:
Where does the magic number come from? Why 1 ms?

Reply:
I noticed from my own manual testing that if we did not sleep, all the files were created at the same instant (as far as the operating system was concerned). Therefore, I could not sort the files by creating time.
If we instead wait any amount of time, even only 1 millisecond, the files could now be sorted.

I picked 1 millisecond because that was the minimum amount of time I could sleep using `await new Promise`. Do you think I should move the magic number to a const? It seems unnecessary since the magic number is 1 and not some complex number.

Reply:
This makes sense, sounds good. Make a `sleep` function for readability!

Reply:
Thanks. I have created a `sleep` function that accepts a parameter `ms`!
```
/**
 * @param ms number of milliseconds to sleep
 */
export const sleep = async (ms: number) => {
  await new Promise((resolve) => setTimeout(resolve, ms))
}
```